### PR TITLE
docs: add example notebook for linearSVR

### PR DESCRIPTION
**Pull request by oboulant**
This notebook introduces an example on how to use linearSVR and a comparison with sklearn.

It is inspired by :

* https://github.com/zama-ai/concrete-ml/blob/main/docs/advanced_examples/LinearRegression.ipynb 
* https://github.com/zama-ai/concrete-ml/blob/release/0.6.x/docs/advanced_examples/XGBClassifier.ipynb
* https://github.com/zama-ai/concrete-ml/blob/release/0.6.x/docs/advanced_examples/DecisionTreeClassifier.ipynb

### CODE REVIEW

**Comment by RomanBredehoft** [comment]
Hello,
Thank a lot for your PR ! We think that the notebook could be improved, so here is a couple of suggestions: 
- You should use our latest public version, CML 0.6.1 ! You'll see that linear models such as the linear SVR one are now much faster compared to what you experienced (for both the inference's execution the key generation times). Also, this makes you able to to use many more number of bits for quantization ( `n_bits` parameter). You pointed out that the default is 8 bits but you could try even more (10 or 16). You'll then see that the model almost exactly matches Scikit-Learn's performance ! 
- You could also try to include this `n_bits` parameter in the GridSearch in order to show that you might actually not need that much (we saw through experimentation that 8 is usually enough)
- The plots are nice but also look a bit too similar to the Linear Regression notebook. It would be great to not showcase only synthetically-generated data sets, could you then maybe try to use a real (non-synthetic) one ? It does not have to be big or complex. Also, suggesting a case where this linear SVR model performs better than a regular linear regression one is only a plus.

Apart from that, the notebook overall is great, you really made the effort to include explanations about our main features and constraints. It has a coherent structure and is easy to follow.

Thanks again !

**Comment by oboulant** [comment]
Thx for the comments, I will wait on the new concrete-ml version to be out to iterate on this. Indeed, I am working on an Apple M2 Silicon and seems like the new version will make it way easier to work on this hardware (especially if I increase `n_bits`). 
I will push something then.

**Comment by aquint-zama** [comment]
concrete-ml v1.0.0 with Apple Silicon support is out

**Comment by oboulant** [comment]
@RomanBredehoft I pushed a new version taking into account your comments. I now used some real data and include `n_bits` into the GridSearch. 

I adjusted the comments accordingly. 

I attempted to compare with a regular OLS but could not find a setup where SVR outperformed. I also tried to deliberately  add an outlier to show that SVR in that case could outperform but chose not to include it since IMO it entropies the notebook and the message (it makes the reader focus on 2 things rather than 1 : {Concrete-ML capabilities for SVR + OLS vs. SVR} vs. {Concrete-ML capabilities for SVR}. 

WDYT ?

**Comment by RomanBredehoft** [comment]
Thanks a lot for your update, it looks great ! Could you maybe run the notebook so that we can see the printed outputs ?

**Comment by oboulant** [comment]
> Thanks a lot for your update, it looks great ! Could you maybe run the notebook so that we can see the printed outputs ?

Yes, just done ⬆️ ! 

Sorry about that, initially I did not want to push anything dependant on runtime / platform, but you are right : for reviewing notebooks it makes it easier :)

**Comment by RomanBredehoft** [comment]
No worries ! The plot made me realize that it would preferable to compare sklearn vs cml models using our `fit_benchmark` method !

So instead of having : 
```
# Train concrete and sklearn LinearSVR  with best hyper parameters
concrete_rgs = ConcreteLinearSVR(**best_params_concrete)
sklearn_rgs = SklearnLinearSVR(**best_params_sklearn)

concrete_rgs.fit_benchmark(X_train, y_train)
sklearn_rgs.fit_benchmark(X_train, y_train)
```
do
```
# Train concrete and sklearn LinearSVR  with best hyper parameters
concrete_rgs = ConcreteLinearSVR(**best_params_concrete)

concrete_rgs, sklearn_rgs = concrete_rgs.fit_benchmark(X_train, y_train)
```

this has the advantage to make sklearn use exactly the same hyper-parameters as the concrete-ml one, and thus better show how the performance are even less impacted by quantization !

**Comment by oboulant** [comment]
Done !

**Comment by RomanBredehoft** [comment]
Thanks a lot, it looks great to me :slightly_smiling_face:  We'll discuss about it with the rest of the team at the beginning of next week and then I'll come back to you !

**Comment by fd0r** [comment]
Hello @oboulant , thanks a lot for your contribution!

Could you now open a PR with the notebook in the folder `docs/advanced_examples` on the branch https://github.com/zama-ai/concrete-ml/tree/tutorials please?
Using [`CapitalizedWords`](https://en.wikipedia.org/wiki/Camel_case) for the name of the notebook.

Once the PR is merged in the branch with the other bounty contributions we will procede to merge it on main and your work will be featured in the next release!

**Comment by oboulant** [comment]
> Hello @oboulant , thanks a lot for your contribution!
> 
> Could you now open a PR with the notebook in the folder `docs/advanced_examples` on the branch https://github.com/zama-ai/concrete-ml/tree/tutorials please? Using [`CapitalizedWords`](https://en.wikipedia.org/wiki/Camel_case) for the name of the notebook.
> 
> Once the PR is merged in the branch with the other bounty contributions we will procede to merge it on main and your work will be featured in the next release!

Thx a lot ! 

Related PR has been created here : https://github.com/zama-ai/concrete-ml/pull/71
